### PR TITLE
use exisiting Search/Add mechanism to edit ins from demographics

### LINKED ISSUE #6082: slow Insurance company load times Admin->Practice->Practice Settings
### Describe the bug

[forum report](https://community.open-emr.org/t/pagination-on-insurance-companies-list/19789)

### Expected behavior
Insurance companies load quickly

### PR DESCRIPTION
**Pull request by stephenwaite**
a way to fix #6082

#### Short description of what this resolves:
Smarty is bogged down by factory methods when there's hundreds of insurance companies to handle. 

#### Changes proposed in this pull request:
add the ability to update insurance companies through the demographics page

### CODE REVIEW

File: interface/practice/ins_search.php [hunk 1]
```diff
@@ -211,42 +249,55 @@ function validate(f) {
 
 <p>
 <table class="w-100 border-0">
+ <tr>
+  <td class="font-weight-bold" width='1%' nowrap><?php echo xlt('Id'); ?>:</td>
+  <td>
+   <input type='text' size='20' name='form_id' maxlength='35' class='form-control form-control-sm' title='<?php echo xla('Id of insurance company'); ?>'
+       readonly='readonly' value='<?php echo text($ins_co['id'] ?? ''); ?>' />
```
Comment (bradymiller) [inline interface/practice/ins_search.php:256]: attr()

File: interface/practice/ins_search.php [hunk 2]
```diff
@@ -211,42 +249,55 @@ function validate(f) {
...
+  </td>
+ </tr>
  <tr>
   <td class="font-weight-bold" width='1%' nowrap><?php echo xlt('Name'); ?>:</td>
   <td>
-   <input type='text' size='20' name='form_name' maxlength='35' class='form-control form-control-sm' title='<?php echo xla('Name of insurance company'); ?>' />
+   <input type='text' size='20' name='form_name' maxlength='35' class='form-control form-control-sm' title='<?php echo xla('Name of insurance company'); ?>'
+       value='<?php echo text($ins_co['name'] ?? ''); ?>' />
```
Comment (bradymiller) [inline interface/practice/ins_search.php:263]: attr()

File: interface/practice/ins_search.php [hunk 3]
```diff
@@ -211,42 +249,55 @@ function validate(f) {
...
   </td>
  </tr>
 
  <tr>
   <td class="font-weight-bold" nowrap><?php echo xlt('Attention');?>:</td>
   <td>
-   <input type='text' size='20' name='form_attn' maxlength='35' class='form-control form-control-sm' title='<?php echo xla('Contact name'); ?>' />
+   <input type='text' size='20' name='form_attn' maxlength='35' class='form-control form-control-sm' title='<?php echo xla('Contact name'); ?>' 
+       value='<?php echo text($ins_co['attn'] ?? ''); ?>' />
```
Comment (bradymiller) [inline interface/practice/ins_search.php:271]: attr() above and lines 279, 287, 296, 300, 310, 314, 323, 335,

File: src/Services/PhoneNumberService.php [hunk 4]
```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ * PhoneService
```
Comment (bradymiller) [inline src/Services/PhoneNumberService.php:4]: dance-bear

**Comment by bradymiller** [comment]
@stephenwaite , just noted some escaping issues. If testing well, then I think is ready to come in after fix those.

**Comment by stephenwaite** [comment]
thanks @bradymiller , this is testing well. There's one little minor thing that happens when you update the insurance name and gets returned to the demos form with the old name. Something to do with select2 trigger but it doesn't have to hold up bringing it in.

**Comment by bradymiller** [comment]
hi @stephenwaite , up to you if bring in or wait :)

**Pull request merged**